source/solvers/heat_transfer.cc

```diff
@@ -217,38 +241,82 @@ HeatTransfer<dim>::assemble_system(
             {
               // Store JxW in local variable for faster access
               const double JxW = fe_values_ht.JxW(q);
+
+              const auto velocity = velocity_values[q];
+
+
+              // Calculation of the magnitude of the velocity for the
+              // stabilization parameter
+              const double u_mag = std::max(velocity.norm(), 1e-12);
+
+              // Calculation of the GLS stabilization parameter. The
+              // stabilization parameter used is different if the simulation is
+              // steady or unsteady. In the unsteady case it includes the value
+              // of the time-step
+              const double tau =
+                is_steady(time_stepping_method) ?
```

Comment:
I didn't know this formulation! Is it faster than a simple if/else loop?

Reply:
It more importantly allow you to declare a variable whose content depend on an if. This way it can be const.